src/rust/engine/fs/store/src/lib.rs

```diff
@@ -1105,7 +1105,11 @@ impl Store {
           // expected to create parent directories for both files and empty leaf
           // directories.)
           if directory.directories.is_empty() {
-            future::ready(Ok(vec![DigestEntry::EmptyDirectory(path_so_far.into())])).boxed()
+            if path_so_far.as_os_str().is_empty() {
```

Comment:
clippy complained about using `!…is_empty()` which makes for this weird pattern of putting the only non-degenerate case in the middle of the three return values. If there's a better solution for this, please let me know :)

Reply:
That is fine. The Clippy must be satisfied. (And not really anything else to do short of disabling the lint which it does not make sense to do here.)

Reply:
@stuhood Thank you for that! I'll try doing that instead.